### ISSUE TITLE
in wilds/hparams_registry.py

### ISSUE THREAD

**Issue by soybean512** [NancyQuris/XDomainMix#3]
wilds/hparams_registry.py
# Unconditional hparam definitions.
    _hparam('warmup_step', 4000, lambda r: 4000)
    _hparam('threshold', 0.9, lambda r: 0.9)
    _hparam('threshold_lower_bound', 0.5, lambda r: 0.5)
    _hparam('probability_to_discard', 0.2, lambda r: r.choice([0.2, 0.3, 0.4]))
    _hparam('value_to_change', 0.1, lambda r: 10**r.uniform(-1, -2))
    _hparam('step_to_change', 500, lambda r: int(50*r.uniform(1, 6)))
    _hparam('rsc_f_drop_factor', 1/3, lambda r: r.uniform(0, 0.5))
    _hparam('rsc_b_drop_factor', 1/3, lambda r: r.uniform(0, 0.5))

    _hparam('rho', 0.05, lambda r: r.choice([0.01, 0.02, 0.05, 0.1]))
    _hparam('alpha', 0.05, lambda r: r.choice([0.01, 0.02, 0.05, 0.1]))
    return hparams

Hello dear Dr. Nancy 
I replaced the camelyon data set in the project with my own data set. I want to know what the Unconditional hparam definitions.in the hparams _ registry.py file in wilds is for the problem of replacing the data set, and how to modify it, especially the warmup _ step stage.

**Comment by NancyQuris** [comment]
Sorry for the misleading comment "unconditional hparam definition" in `hparams_registry.py`. Please ignore it. 

The hyperparameters you listed above are the those used by our method and methods that we implemented. 

Specifically, 
```
_hparam('warmup_step', 4000, lambda r: 4000)
_hparam('threshold', 0.9, lambda r: 0.9)
_hparam('threshold_lower_bound', 0.5, lambda r: 0.5)
_hparam('probability_to_discard', 0.2, lambda r: r.choice([0.2, 0.3, 0.4]))
_hparam('value_to_change', 0.1, lambda r: 10**r.uniform(-1, -2))
_hparam('step_to_change', 500, lambda r: int(50*r.uniform(1, 6)))
```
are the hyperparameters used by XDomainMix. You can refer to our paper for details about the hyperparameters. You can tune them to see how they affect the performance on a new dataset.

XDomainMix will only be conducted after the warmup stage, as the feature extractor, class classifier and domain classifier need to be adequately trained to extract meaningful features and identify class/domain-related components. The stopping crite-
ria for the warm-up period can be determined by the number of training steps needed for both the accuracy and loss function to stabilize.
